xc-tool/build.rs

```diff
@@ -0,0 +1,107 @@
+use std::{env, fs, path::PathBuf, path::Path};
+
+use ethers::contract::Abigen;
+use ethers::solc::{remappings::Remapping, Artifact, Project, ProjectPathsConfig, Solc};
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
We should use the `anyhow::Result` type everywhere, it makes dealing with errors from different crates much easier.

Reply:
I can add it, but I have my doubts.

I don't think it's `anyhow`'s fault, I think error crates are still vastly immature in Rust - in general. I can count at least three  U L T I M A T E™ and  D E F I N I T I V E™ crates that were the premier choice over the years. I'm talking about `failure`, then recently `anyhow` and most recently `eyre` seems to be the talk of the town. I'm sure they're awesome, but time and again, the boxed std trait object seemed to be the common denominator, supported by default, working in 80% of the cases. `build.rs` files are meant to be more of a script affair and I'm not convinced of the quality-of-life necessity here.